yaml_test.go

```diff
@@ -17,16 +17,11 @@ limitations under the License.
 package yaml
 
 import (
-	"encoding/json"
 	"fmt"
 	"math"
 	"reflect"
-	"sort"
 	"strconv"
 	"testing"
-
-	"github.com/davecgh/go-spew/spew"
-	yaml "gopkg.in/yaml.v2"
 )
 
 /* Test helper functions */
```

Comment:
In the JSONToYAML and Marshal tests I would expect the sequence indentation style to have changed to wide instead of compact, due to an upstream change. Is that true? As discussed earlier, kustomize carries a fork of go-yaml for that reason, to provide customizability.

Reply:
That upstream change blocks migration to yaml.v3. We don't want any externally visible change in serialization behavior.

Reply:
Thanks for the information, I'll probably create a fork of go-yaml in this repo too & create an upstream PR (if no PR is  present yet).

Reply:
I don't think we're open to forking the yaml repo at this point. The kustomize fork is intended to be temporary and they are pursuing options to bring the V3 library back into a compatible state

Reply:
the fixes kustomize needs to land in yaml.v3 to restore compatibility are either in https://github.com/go-yaml/yaml/pull/756 or https://github.com/go-yaml/yaml/pull/753

Reply:
Ok, but can't we add a temporary fork here too? Seems like the kyaml team has been trying for a long time to get this option added. Would be great if we could not block this PR until upstream has fixed the issue.

Reply:
we could use yaml.v3 for unmarshalling and yaml.v2 for marshalling